### PR TITLE
Add nodemetadata debug interface

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

Allows users to get proxy node metadata via /debug/adsz

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/xds/debug.go [hunk 1]
```diff
@@ -97,6 +97,7 @@ type AdsClient struct {
 	ConnectionID string              `json:"connectionId"`
 	ConnectedAt  time.Time           `json:"connectedAt"`
 	PeerAddress  string              `json:"address"`
+	Metadata     *model.NodeMetadata `json:"metadata"`
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/debug.go:100]: This is the change

**Comment by hzxuzhonghu** [comment]
This can help check metadata, especially vm proxy.

File: pilot/pkg/xds/debug.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/xds/debug.go:100]: would prefer to isolate the refactor

File: pilot/pkg/xds/debug.go [hunk 2]
```diff
@@ -529,15 +530,25 @@ func (s *DiscoveryServer) adsz(w http.ResponseWriter, req *http.Request) {
 	if s.handlePushRequest(w, req) {
 		return
 	}
+	connections := s.Clients()
```
Comment (howardjohn) [inline pilot/pkg/xds/debug.go:533]: nit: if we hit L535 its unnecessary to all `.Clients()`, throwaway work

File: pilot/pkg/xds/debug.go [hunk 3]
```diff
@@ -529,15 +530,25 @@ func (s *DiscoveryServer) adsz(w http.ResponseWriter, req *http.Request) {
...
+	if proxyID := req.URL.Query().Get("proxyID"); proxyID != "" {
```
Comment (howardjohn) [inline pilot/pkg/xds/debug.go:534]: use `s.getDebugConnection` maybe to simplify?

Comment (howardjohn) [reply pilot/pkg/xds/debug.go:534]: We can modify it to make it not fail on missing proxyID query param if needed?

**Review by howardjohn — APPROVED** [review]
LGTM, few small nits

File: pilot/pkg/xds/debug.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/xds/debug.go:100]: Got it

**Pull request merged**